src/main/java/dev/jbang/util/WinJunction.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2011 杨博 (Yang Bo)
+ * Translated from Scala to Java 2024 (Tako Schotanus)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.jbang.util;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import com.sun.jna.Memory;
+import com.sun.jna.Native;
+import com.sun.jna.platform.win32.WinBase;
+import com.sun.jna.platform.win32.WinError;
+import com.sun.jna.platform.win32.WinNT;
+import com.sun.jna.platform.win32.WinNT.HANDLE;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.win32.StdCallLibrary;
+import com.sun.jna.win32.W32APIOptions;
+
+public class WinJunction {
+
+	private static final int sizeOfReparseDataHead = 8;
+	private static final int sizeOfMountPointReparseHead = 8;
+	private static final int IO_REPARSE_TAG_MOUNT_POINT = 0xA0000003;
+	private static final int FSCTL_SET_REPARSE_POINT = 0x000900a4;
+
+	private static final String PathPrefix = "\\??\\";
+
+	public interface Kernel32 extends StdCallLibrary {
```

Comment:
will this work in newer java versions too without warning/errors?

Reply:
Why wouldn't it? Is there something here I'm not seeing?

Reply:
i might remember wrong but just have recollection that jna library is't fully supported across java 8 to Java 23+ but it might just be a rumour.